packages/api-v4/src/images/sharegroup.ts

```diff
@@ -0,0 +1,237 @@
+import {
+  addSharegroupMemberSchema,
+  generateSharegroupTokenSchema,
+  updateSharegroupMemberSchema,
+  updateSharegroupTokenSchema,
+} from '@linode/validation/lib/images.schema';
+
+import { BETA_API_ROOT } from '../constants';
+import Request, {
+  setData,
+  setMethod,
+  setParams,
+  setURL,
+  setXFilter,
+} from '../request';
+
+import type { Filter, ResourcePage as Page, Params } from '../types';
+import type {
+  AddSharegroupMemberPayload,
+  GenerateSharegroupTokenPayload,
+  Image,
+  Sharegroup,
+  SharegroupMember,
+  SharegroupToken,
+  UpdateSharegroupMemberPayload,
+} from './types';
+
+/**
+ *  Add Member to the Sharegroup
+ *
+ * @param sharegroupId {string} ID of the Sharegroup to add member
+ * @param data {AddSharegroupMemberPayload} the Member details
+ */
+export const addMembersToSharegroup = (
+  sharegroupId: number,
+  data: AddSharegroupMemberPayload,
+) => {
+  return Request<Sharegroup>(
+    setURL(
+      `${BETA_API_ROOT}/images/sharegroups/${encodeURIComponent(sharegroupId)}/images`,
+    ),
+    setMethod('POST'),
+    setData(data, addSharegroupMemberSchema),
+  );
+};
+
+/**
+ * Generate user token for the Sharegroup
+ *
+ * @param data {GenerateSharegroupTokenPayload} the token details
+ */
+export const generateSharegroupToken = (
+  data: GenerateSharegroupTokenPayload,
+) => {
+  return Request<SharegroupToken>(
+    setURL(`${BETA_API_ROOT}/images/sharegroup/tokens`),
```

Comment:
```suggestion
    setURL(`${BETA_API_ROOT}/images/sharegroups/tokens`),
```
I think this should be `sharegroups` (per the specs). Could you double-check?